build: pass the framework path to the importer

compnerd opened the pull request:
This is to silence the warning when building the lldb test suite.

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

compnerd commented:
CC: @dcci @gottesmm

compnerd commented:
@swift-ci please test and merge

jrose-apple reviewed (dismissed):
-F flags already get passed through to Clang, so this can't be the right fix.

compnerd committed:
build: adjust rpath, don't pass `-F` to the link

Pass the framework only to the importer as `-F` on ELF is used for the
`DT_FILTER`.  Adjust the `DT_RUNPATH` so that the dependent libraries are found.

jrose-apple commented:
That sounds like a Clang bug, since Clang is the linker. Right?

compnerd commented:
@jrose-apple - fair enough.  I don't have a problem with fixing that in clang, but, I think that we should at least do this as a temporary thing.

jrose-apple commented:
That makes sense. If you end up making more changes, then, mind inserting a comment to that effect?

Thanks for the explanation!

compnerd commented:
@swift-ci please clean test macOS platform
